packages/finance/src/liability/liquidation.rs

```diff
@@ -85,85 +85,93 @@ where
 {
     debug_assert!(total_due <= asset);
     debug_assert!(overdue <= total_due);
-    may_ask_liquidation_liability(spec, asset, total_due, min_asset)
-        .max(may_ask_liquidation_overdue(
-            asset,
-            overdue,
-            min_liquidation,
-            min_asset,
-        ))
-        .unwrap_or_else(|| no_liquidation(spec, asset, total_due))
+    may_ask_liquidation_liability(spec, asset, total_due, min_liquidation, min_asset).max(
+        may_ask_liquidation_overdue(spec, asset, overdue, min_liquidation, min_asset),
+    )
 }
 
-fn no_liquidation<Asset>(
+fn may_ask_liquidation_liability<Asset>(
     spec: &Liability,
     asset: Coin<Asset>,
     total_due: Coin<Asset>,
+    min_liquidation: Coin<Asset>,
+    min_asset: Coin<Asset>,
 ) -> Status<Asset>
 where
     Asset: Currency,
 {
-    if total_due.is_zero() {
-        Status::NoDebt
+    let liquidation_amount = spec.amount_to_liquidate(asset, total_due);
+    let ltv = ltv(total_due, asset);
+    if liquidation_amount < min_liquidation {
+        if ltv >= spec.max() {
+            no_liquidation(spec, total_due, spec.third_liq_warn())
+        } else {
+            no_liquidation(spec, total_due, ltv)
+        }
     } else {
-        let ltv = Percent::from_ratio(total_due, asset);
-        debug_assert!(ltv < spec.max());
-
-        Status::No(spec.zone_of(ltv))
+        ask_liquidation(
+            asset,
+            Cause::Liability {
+                ltv: spec.max(),
+                healthy_ltv: spec.healthy_percent(),
+            },
+            liquidation_amount,
+            min_asset,
+        )
     }
 }
 
-fn may_ask_liquidation_liability<Asset>(
+fn may_ask_liquidation_overdue<Asset>(
     spec: &Liability,
     asset: Coin<Asset>,
-    total_due: Coin<Asset>,
+    overdue: Coin<Asset>,
+    min_liquidation: Coin<Asset>,
     min_asset: Coin<Asset>,
-) -> Option<Status<Asset>>
+) -> Status<Asset>
 where
     Asset: Currency,
 {
-    may_ask_liquidation(
-        asset,
-        Cause::Liability {
-            ltv: spec.max(),
-            healthy_ltv: spec.healthy_percent(),
-        },
-        spec.amount_to_liquidate(asset, total_due),
-        min_asset,
-    )
+    let ltv = ltv(overdue, asset);
+    if overdue < min_liquidation {
+        no_liquidation(spec, overdue, ltv)
+    } else {
+        ask_liquidation(asset, Cause::Overdue(), overdue, min_asset)
+    }
```

Comment:
this logic is similar to the one used on liquidation by liability ... think on reusing it